### PR TITLE
xinetd: Add xinetd package

### PR DESCRIPTION
**Pull request by terasawa-lin**
Modifications to the recipe
add:
inherit gettext for configure
Run autogen.sh on do_configure
Added parameters to oe_runconf
do_compile
RDEPEND

remove:
SRC_URI
S

change:
reference directory for do_install

# Purpose of pull request

This pr adds xinetd package based on poky/meta/recipes-extended/xinetd/xinetd_2.3.15.bb

# Test
## How to install
Add the following to conf/local.conf.
```
IMAGE_INSTALL_append = " xinetd"
```

## How to test
Run following command from console

```
# vi /etc/xinetd.d/echo
disable = yes
↓
disable = no

# /etc/init.d/xinetd restart
```

## Test result
If you connect to the port number 7 of the local host with the nc command and enter a character, the entered character is returned.
```
# nc localhost 7
aaa
aaa
^C
```

### CODE REVIEW

File: recipes-debian/xinetd/xinetd_debian.bb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# Apple and Gentoo say BSD here.
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://COPYRIGHT;md5=55c5fdf02cfcca3fc9621b6f2ceae10f"
+
+DEPENDS += "libtirpc"
+RDEPENDS_${PN} = "perl lsb perl-module-socket perl-module-io-socket "
+PR = "r2"
+
+# Blacklist a bogus tag in upstream check
+UPSTREAM_CHECK_GITTAGREGEX = "xinetd-(?P<pver>(?!20030122).+)"
```
Comment (hiraku-wfs) [inline recipes-debian/xinetd/xinetd_debian.bb]: We use Debian source package, so we don't need this.

File: recipes-debian/xinetd/xinetd_debian.bb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+DEBIAN_PATCH_TYPE = "quilt"
+DEBIAN_QUILT_PATCHES = ""
+
+FILESPATH_append = ":${COREBASE}/meta/recipes-extended/xinetd/xinetd"
+
+UPSTREAM_VERSION_UNKNOWN = "1"
```
Comment (hiraku-wfs) [inline recipes-debian/xinetd/xinetd_debian.bb]: Is this necessary?

File: recipes-debian/xinetd/xinetd_debian.bb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+SRCREV = "68bb9ab9e9f214ad8a2322f28ac1d6733e70bc24"
```
Comment (hiraku-wfs) [inline recipes-debian/xinetd/xinetd_debian.bb]: We use Debian source package, so we don't need this.

**Comment by terasawa-lin** [comment]
Fixed for 3 comments.

File: recipes-debian/xinetd/xinetd_debian.bb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://COPYRIGHT;md5=55c5fdf02cfcca3fc9621b6f2ceae10f"
+
+DEPENDS += "libtirpc"
+RDEPENDS_${PN} = "perl lsb perl-module-socket perl-module-io-socket "
+
+DEBIAN_PATCH_TYPE = "quilt"
+DEBIAN_QUILT_PATCHES = ""
+
+S = "${WORKDIR}/${BP}"
```
Comment (masami256) [inline recipes-debian/xinetd/xinetd_debian.bb]: ```DEBIAN_UNPACK_DIR``` is same as ``` "${WORKDIR}/${BP}"``` and ```S``` will be set  as ```DEBIAN_UNPACK_DIR```. so you don't need to set S here.

**Comment by terasawa-lin** [comment]
変数Sを取り除きました。

**Comment by terasawa-lin** [comment]
Removed the variable S.

**Comment by masami256** [comment]
Confirmed that xinet.d works both sysvinit and systemd fine.

**Pull request merged**